src/hooks/useApeModeQueryParamReader.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react'
+import { useDispatch } from 'react-redux'
+import { AppDispatch } from 'state'
+import { updateUserExpertMode } from '../state/user/actions'
+import useParsedQueryString from './useParsedQueryString'
+
+export default function ApeModeQueryParamReader(): null {
+  useApeModeQueryParamReader()
+  return null
+}
+
+function useApeModeQueryParamReader() {
+  const dispatch = useDispatch<AppDispatch>()
+  const { ape } = useParsedQueryString()
+
+  useEffect(() => {
+    if (typeof ape !== 'string') return
+
+    dispatch(updateUserExpertMode({ userExpertMode: ape.toLowerCase() === 'true' }))
```

Comment:
i think we should only dispatch this if `ape` is undefined or equal to `true`, otherwise do not dispatch an action that disables expert mode

Reply:
should ?ape and ?ape= enable expert mode?